crates/hir-def/src/body.rs

```diff
@@ -257,6 +270,17 @@ impl Body {
         f(pat_id);
         self.walk_pats_shallow(pat_id, |p| self.walk_pats(p, f));
     }
+
+    pub fn is_binding_upvar(&self, binding: BindingId, relative_to: ExprId) -> bool {
+        match self.binding_owners.get(&binding) {
+            Some(x) => {
+                // We assign expression ids in a way that outer closures will receive
+                // a lower id
+                x.into_raw() < relative_to.into_raw()
+            }
+            None => true,
```

Comment:
@HKalbasi this is unchanged, but shouldn't the `None` branch here return `false`? Or am I misunderstanding this function

Reply:
In case of `None` the binding is defined in the root of body, so it is upvar to any closure.

Reply:
Ah, okay I was misunderstanding then, thanks